src/containers/ListingPage/ListingPage.duck.js

```diff
@@ -406,10 +410,14 @@ export const loadData = (params, search, config) => (dispatch, getState, sdk) =>
     return Promise.resolve();
   }
 
-  return Promise.all([
-    dispatch(showListing(listingId, config)),
-    dispatch(fetchReviews(listingId)),
-  ]).then(response => {
+  const hasViewingRights = currentUser && hasReadPermission(currentUser);
+  const promises = hasViewingRights
+    ? // For users with viewing rights, fetch the listing and the associated reviews
+      [dispatch(showListing(listingId, config)), dispatch(fetchReviews(listingId))]
+    : // If user has no viewing rights, only allow fetching their own listing without reviews
+      [dispatch(showListing(listingId, config, true))];
+
+  return Promise.all(promises).then(response => {
```

Comment:
You need to restrict also fetchMonthlyTimeSlots call that happens in the "then" scope.

Reply:
Fixed!